engine/src/main/java/org/hibernate/validator/internal/engine/ConfigurationImpl.java

```diff
@@ -494,23 +498,28 @@ private void parseValidationXml() {
 			);
 			applyXmlSettings( xmlParameters );
 		}
-
-		for ( ValueExtractorDescriptor valueExtractorDescriptor : valueExtractorDescriptors.values() ) {
-			validationBootstrapParameters.addValueExtractorDescriptor( valueExtractorDescriptor );
-		}
 	}
 
 	private void loadValueExtractorsFromServiceLoader() {
-		@SuppressWarnings("rawtypes")
-		ServiceLoader<ValueExtractor> loader = ServiceLoader.load(
-				ValueExtractor.class,
-				externalClassLoader != null ? externalClassLoader : getClass().getClassLoader()
-		);
+		List<ValueExtractor<?>> valueExtractors;
+
+		if ( externalClassLoader != null ) {
+			final ClassLoader originalContextClassLoader = run( GetClassLoader.fromContext() );
```

Comment:
Mmmmh, OK, I must admit I wanted to be on par with the patterns we used elsewhere.